qiskit/utils/deprecation.py

```diff
@@ -17,10 +17,15 @@
 from typing import Type
 
 
-def deprecate_arguments(kwarg_map, category: Type[Warning] = DeprecationWarning):
+def deprecate_arguments(
+    kwarg_map, category: Type[Warning] = DeprecationWarning, docstring_version=None
```

Comment:
Minor nitpick, but I wouldn't be opposed to calling the new argument `version` or `since` instead, since we might want to change the exact effects of the function in the future.  It would also be good to make it non-optional, but we'd have to make sure all the downstream packages are ready for that change first.

Reply:
The idea is to skip the docstring annotation by `docstring_version=None`. In this way, the dev could implement their our annotation.

Reply:
I feel like that would be better as a separate Boolean flag - make `since`/`version` required (that way we're guaranteed to get your nice searching, too), and have a separate flag `modify_docstring=True`?

Reply:
done